cwltool/provenance.py

```diff
@@ -324,13 +322,13 @@ def __init__(self,
         self.user_provenance = user_provenance
         self.engine_uuid = research_object.engine_uuid
         self.add_to_manifest = self.research_object.add_to_manifest
-        if self.orcid:
+        if self.orcid is not None:
             _logger.debug(u"[provenance] Creator ORCID: %s", self.orcid)
         self.full_name = full_name or None
-        if self.full_name:
+        if self.full_name is not None:
```

Comment:
Empty "full name" is non-nonsensical and should not be logged in provenance as such. 

While perhaps undocumented, setting an empty name can in the old code be used on command line to override `CWL_FULL_NAME` environment variable, as it's not possible to say `--full-name=None` on the command line.

Reply:
Since it doesn't seem to be a difference between the usage if `None` and `''` as they should both be ignore, I'm leaning towards removing the value `None` from the code:
1. Set the default value to `''` instead of `None`
2. Don't allow the variable to be None

I'm not seeing any tests dealing with the name, could you give some hints on how to check normal behaviour? I might be able to create some corner cases from there.

Reply:
Look at the following options for `cwltool`:
```
Options for recording provenance information of the execution:
  --provenance PROVENANCE
                        Save provenance to specified folder as a Research
                        Object that captures and aggregates workflow execution
                        and data products.
  --enable-user-provenance
                        Record user account info as part of provenance.
  --disable-user-provenance
                        Do not record user account info in provenance.
  --enable-host-provenance
                        Record host info as part of provenance.
  --disable-host-provenance
                        Do not record host info in provenance.
  --orcid ORCID         Record user ORCID identifier as part of provenance,
                        e.g. https://orcid.org/0000-0002-1825-0097 or
                        0000-0002-1825-0097. Alternatively the environment
                        variable ORCID may be set.
  --full-name CWL_FULL_NAME
                        Record full name of user as part of provenance, e.g.
                        Josiah Carberry. You may need to use shell quotes to
                        preserve spaces. Alternatively the environment
                        variable CWL_FULL_NAME may be set.
```

Reply:
I was more worried about where is the result stored:

I'm seeing 21 hits across 9 files when I set a full-name, I wanted to know what is the expected result when the name is blank, are these fields written? if not, how so?